hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/validate/HazelcastSqlValidator.java

```diff
@@ -133,6 +143,18 @@ public SqlNode validate(SqlNode topNode) {
             return topNode;
         }
 
+        if (topNode instanceof SqlExplainStatement) {
+            SqlExplainStatement explainStatement = (SqlExplainStatement) topNode;
+            SqlNode explicandum = explainStatement.getExplicandum();
+
+            if (explicandum instanceof SqlOrderBy) {
+                explicandum = super.performUnconditionalRewrites(explicandum, false);
```

Comment:
1. Why do we explicitly call the superclass' impl? It's unusual except for methods calling their own super implementation. 
2. Why do we do the rewrite only for `SqlOrderBy`? Calcite does it as a part of validation for each query, so should we for EXPLAIN. Or maybe we should call the validation?

`ExplainStatementTest` passes if I do both changes above.

Reply:
It works this way: 
```java
        if (topNode instanceof SqlExplainStatement) {
            SqlExplainStatement explainStatement = (SqlExplainStatement) topNode;
            SqlNode newExplicandum = validate(explainStatement.getExplicandum());
            explainStatement.setExplicandum(newExplicandum);
            return explainStatement;
        }
```
I was experimenting also with removing this whole `if`. But I ran into issues and I gave up. Conceptually I think it should work, but there's some small difference that breaks it...

Reply:
> Why do we do the rewrite only for SqlOrderBy?

It's optimisation corner case, I guess. `SqlOrderBy` is present as AST node (or `SqlNode`), but then it become embedded as part of `SqlSelect`, and node itself is removed in `performUnconditionalRewrites()`.  As a result, `ORDER BY` is absent as operator on the next phases and also doesn't present in `SUPPORTED_KINDS`.

Explain query contains explicandum query, and  `performUnconditionalRewrites()` doesn't rewrite anything for EXPLAIN. It's a reason we do it manually.